specs/dotsigstore_bundle/pb/sigstore_bundle.proto

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.bundle.v1;
+
+// https://raw.githubusercontent.com/secure-systems-lab/dsse/9c813476bd36de70a5738c72e784f123ecea16af/envelope.proto
+import "envelope.proto";
+import "sigstore_common.proto";
+import "sigstore_rekor.proto";
+
+option go_package = "github.com/sigstore/proto/bundle/v1";
+option java_package = "dev.sigstore.proto.bundle.v1";
+option java_multiple_files = true;
+option java_outer_classname = "BundleProto";
+
+// Notes on versioning.
+// The primary message ('Bundle') MUST be versioned, by populating the
+// 'media_type' field. Semver-ish (only major/minor versions) scheme MUST
+// be used. The current version as specified by this file is:
+// application/vnd.dev.sigstore.bundle+json;version=0.1
+// The semantic version is thus '0.1'.
+
+// This message holds different constructs that captures the timestamp
+// of the time the signature was generated.
+// As this message can be either empty (no timestamps), or a combination of
+// an arbitrarily number of transparency log entries and signed timestamps,
+// it is the client's responsibility to implement any required verification
+// policies.
+// When using RFC3161 timestamps together with DSSE envelopes, the message
+// imprint (message digest) MUST be computed over the payload (octets).
+message TimestampVerificationData {
```

Comment:
Yes, I hade a first attempt on verification in this PR but removed now (it's on this branch: https://github.com/kommendorkapten/cosign/blob/bundle_verification/specs/dotsigstore_bundle/verify.md)
As I would think the verification is a larger topic to discuss, I would prefer to have that in a separate PR.

Reply:
> Things like "what if the KindVersions differ between tlog entries"?
> 
> Actually, come to think of that—is that allowed?

IMHO this is not a real concern, today the bundle is only possible to map to two Rekor types: `inoto` and `hashedrecord`. For the verification to succeed an asset in the bundle must be able to correctly be put on different logs with different types, which I would say is not possible. Let's assume an in-toto attestation is placed on two logs, the first as an in-toto attestation, and for the second the hash digest of the envelope is calculated and then placed as a hashed record. In the bundle, the asset is stored as an DSSE Envelope. 
During verification the sencond log's entry would fail, as the code that will reconstruct the hashed record entry would fail, as the asset is stored as DSSE envelope, not a message signature as it would expect. If the asset is stored as a hashed record, the verifier would so fail to reconstruct the DSSE envelope.

So this is now implicitly verified, but when talking about verification flows, we can make this explicit, that the verifier MUST verify that all entries are of the same type. How about same type with different versions? I think we should start with require the same version too, until we know of any concrete use-case where it may make sense to allow that.